url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/view/WebControl.java

```diff
@@ -0,0 +1,434 @@
+			
+		} catch (JSONException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
+		
+		
+		
+		
+		// Finally make the schedule based on the prefs
+		SchedulerFactory schedFac = new SchedulerFactory();
+		Scheduler scheduler = schedFac.createSceduler();
+		try {
+			scheduler.addSubscriber(this);
+		} catch (RemoteException e1) {
+			// TODO Auto-generated catch block
+			e1.printStackTrace();
+		}
+		
+		t = new Thread(scheduler);
+		scheduler.setPref(prefs);
+		t.start();
+		
+		while(t.isAlive())
+		{
+			
+		}
+		
+		
+		Iterator<Course> it = retSched.getCourses().iterator();
+		
+		//while (it.hasNext())
+		//{
+			JSONArray arData = new JSONArray();
+			
+			schedule.put(arData);
+			
+			arData.put(retSched.getCourses());
+			
+			
+			
+			try {
+				System.out.println(arData.getJSONArray(0).toString());
+				
+			} catch (JSONException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			
+			// add the day data
+			Iterator<Course> it1 = retSched.getCourses().iterator();
+			int count = 0;
+			while(it1.hasNext())
+			{
+				Course c = it1.next();
+				ArrayList<Day> days = (ArrayList<Day>) c.getValue(CourseData.DAY.toString());
+				ArrayList<String> dayStrings = new ArrayList<String>();
+				for (int i = 0; i < days.size(); i++)
+				{
+					dayStrings.add(days.get(i).toString());
+				}
+				
+				try {
+					
+					schedule.getJSONArray(0).getJSONArray(0).getJSONObject(count).getJSONObject("courseInfo").remove("Day");//put(7, ls);
+					schedule.getJSONArray(0).getJSONArray(0).getJSONObject(count).getJSONObject("courseInfo").accumulate("Day", dayStrings);//put(7, ls);
+				} catch (JSONException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+				
+				count++;
+			}
+			
+			try {
+				System.out.println(arData.getJSONArray(0).toString());
+			} catch (JSONException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		
+
+		return schedule;
+	}
+
+
+
+
+
+
+
+	// I need to be able to return a string containing a list of course names given a string containing the semester
+
+	public JSONArray getCourseNames(JSONObject Jsemester)
+	{
+		JSONArray names = new JSONArray();
+		//String semester = "";
+
+		CourseCollectionFactory fac = new CourseCollectionFactory();
+		CourseCollectionDB db = fac.createCourseCollection();
+
+		String semester = "";
+		try {
+			semester = Jsemester.getJSONArray("Semester").getString(0);
+
+		} catch (JSONException e1) {
+			// TODO Auto-generated catch block
+			e1.printStackTrace();
+		}
+
+		Collection c = null;
+		try {
+			db.setSemester(semester);
+			c = db.getAllCourseNames();
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+
+		System.out.println(semester);
+
+		try {
+			names.put(0, c);
+			System.out.println(names.getJSONArray(0));
+		} catch (JSONException e1) {
+			// TODO Auto-generated catch block
+			e1.printStackTrace();
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 322:328, 357:363.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.